### PR TITLE
Fix some clippy warnings

### PR DESCRIPTION
**Pull request by remilauzier**

### CODE REVIEW

File: src/base/matrix.rs [hunk 1]
```diff
@@ -357,8 +357,10 @@ mod rkyv_impl {
 }
 
 impl<T, R, C, S> Matrix<T, R, C, S> {
-    /// Creates a new matrix with the given data without statically checking that the matrix
-    /// dimension matches the storage dimension.
+    /// Creates a new matrix with the given data.
+    ///
+    /// # Safety
+    /// The function doesnt verify that the dimension of the matrix correspond to the storage dimension.
```
Comment (filnet) [inline src/base/matrix.rs:363]: typo: doesn't

File: src/base/matrix_slice.rs [hunk 2]
```diff
@@ -29,7 +29,10 @@ macro_rules! slice_storage_impl(
         {}
 
         impl<'a, T, R: Dim, C: Dim, RStride: Dim, CStride: Dim> $T<'a, T, R, C, RStride, CStride> {
-            /// Create a new matrix slice without bound checking and from a raw pointer.
+            /// Create a new matrix slice..
+            ///
+            /// # Safety
+            /// The function doesnt do bound checking and use a raw pointer.
```
Comment (filnet) [inline src/base/matrix_slice.rs:35]: typo: doesn't

Comment (filnet) [reply src/base/matrix_slice.rs:35]: typo: uses

File: examples/unit_wrapper.rs [hunk 3]
```diff
@@ -23,5 +23,5 @@ fn main() {
     let l1 = length_on_direction_with_unit(&v, &Vector3::y_axis());
     let l2 = length_on_direction_without_unit(&v, &Vector3::y());
 
-    assert_eq!(l1, l2)
+    assert!((l1 - l2).abs() < f32::EPSILON)
```
[the remark this answers is not in the document]
Comment (sebcrozet) [reply examples/unit_wrapper.rs:26]: The whole point of this example is to show that we get the exact same result with the `Unit<Vector>` and the `Vector` (both having the exact same components).

**Review by sebcrozet — APPROVED** [review]

**Pull request merged**